crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1915,12 +1927,21 @@ impl<'db> TypeInferenceBuilder<'db> {
 
         if let Some(value) = value.as_deref() {
             let value_ty = self.infer_expression(value);
-            self.add_declaration_with_binding(
-                assignment.into(),
-                definition,
-                annotation_ty,
-                value_ty,
-            );
+            if self.file().is_stub(self.db().upcast()) {
+                self.add_declaration_with_binding(
+                    assignment.into(),
+                    definition,
+                    annotation_ty,
+                    Type::Unknown,
+                );
```

Comment:
This was a bit confusing. Using `add_declaration` here fails because this assignment is a binding so the get here fails:
https://github.com/Glyphack/ruff/blob/9963252dfc51e08a53a1047344bd44cde1493d42/crates/red_knot_python_semantic/src/types/infer.rs#L839

This passes the test but can it possibly break some other behavior?